File: Sorting/selection_sort/algo-1/1-1.py
# Implementing Selection Sort in Pyhton using Basic Synatx using range() & numpy.arange()
# Importing Packages
# import numpy for arange() method
import numpy as np
# Input the Size Of Array
N = int(input())
# Declare a empty List to store elements
A = []
# To read the elements
for i in range(N):
     # Adding element in list using list.append()
     A.append(int(input()))

# Selection Sort Algorithm
# we can also use numpy.arange()
for i in range(N-1):
        # Let i-th element be minimum
        key = i;	
        for j in np.arange(i+1,N): 
            if A[j] < A[key]:
                        key = j
                
        # swap(A[i],A[key])
        temp = A[i];
        A[i] = A[key];
        A[key] = temp;	

# Output
for i in range(N):
    print(A[i],end=" ")

File: Sorting/selection_sort/algo-1/1-1.c
// Implementing Selection Sort in C using Dynamic Memory Allocation
// Header Files
// Preprocessor Directive
#include<stdio.h>
#include<malloc.h>
// Main Calling Function
int main(){
	// Varaible Declaration
	int *A,N,i,j,key;
	// temp varaibel for swapping
	int temp;
	// Input the Size of Array
	scanf("%d",&N);
	// Dynamic Memory Allocation using malloc()
	A = (int*)malloc(N*sizeof(int));
	// Input the Array elements
	for(i=0;i<N;i++){
		scanf("%d",&A[i]);
	}	
	// Selection Sort Algorithm
	// Sorted   : A[0...i-1] 
	// UnSorted : A[i...N-1]
	for(i=0;i<N-1;i++){
		// Let i-th element be minimum
		key = i;	
		for(j=i+1;j<N;j++){
			if(A[j]<A[key]){
				key = j;
			}
		}
		// swap(A[i],A[key])
		temp = A[i];
		A[i] = A[key];
		A[key] = temp;	
	}
	// Output
	for(i=0;i<N;i++){
		printf("%d ",A[i]);
	}
return 0;
}
